shared/actions/chat.js

```diff
@@ -1011,6 +1012,7 @@ function _threadToPagination (thread) {
   }
 }
 
+// enforce no key collisions
```

Comment:
So its still possible to have collisions but i think hard for us to fix this with our existing code. I've asked mike to plumb through timetstamps on errors also. The problem is when we get an error since we don't have a timestamp we use Date.now() which means any errors anywhere in the stream can have the time of now(). This means the timestamp injector (which pulls the timestamp of the message below it) can be 'infected' by this error. We dont' filter this out in the reducer correctly (but the times are wrong anyways) so we sould really just wait to get this plumbed through. 

to repro this easily just put in

```
// TEMP
if (Math.random() > 0.57) {

  message.state = LocalMessageUnboxedState.error
    message.error = {
      errType:LocalMessageUnboxedErrorType.badversion,
    }

}
//

```

in _unboxedToMessage